src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -319,34 +318,57 @@ RESULT_CODE ProtocolPacket::ProtocolHeaderValidator::validate(
   // For Control frames Data Size value shall be less than MTU header
   // For Single and Consecutive Data Size value shall be greater than 0x00
   // and shall be less than payload size
-  if (header.dataSize > payload_size) {
-    LOG4CXX_WARN(logger_,
-                 "Packet data size is "
-                     << header.dataSize
-                     << " and bigger than allowed payload size " << payload_size
-                     << " bytes");
-    return RESULT_FAIL;
-  }
+  // For First Frame Data Size shall be equal 0x08 (payload of this packet will
+  // be 8 bytes).
+
   switch (header.frameType) {
     case FRAME_TYPE_SINGLE:
     case FRAME_TYPE_CONSECUTIVE:
-      if (header.dataSize <= 0u) {
+      if (header.dataSize <= 0u || header.dataSize > payload_size) {
+        LOG4CXX_WARN(
+            logger_,
+            "Packet data size of "
+                << (header.frameType == FRAME_TYPE_SINGLE
+                        ? "FRAME_TYPE_SINGLE"
+                        : "FRAME_TYPE_CONSECUTIVE")
+                << " frame must be in range (0, payload_size=" << payload_size
+                << "], but actual value is " << header.dataSize);
+        return RESULT_FAIL;
+      }
+      break;
+      
+    case FRAME_TYPE_CONTROL:
```

Comment:
@KVGrygoriev I think case for FRAME_TYPE_CONTROL could be merged with a previous one
For writing stringified frame type in the logger you could create a free function in the anonymous namespace which will return a string value depending on `frameType` value

Reply:
@AKalinich-Luxoft good point, see 48a5b59baad16acf483c8178829eccd346796636